### ISSUE TITLE
Clear Session on HTTP 401 Response

### ISSUE THREAD

**Issue by eric-horodyski** [ionic-enterprise/tea-taster-react#23]
One piece of functionality across the Tea Taster applications is handling an HTTP 401 Response and clearing the user's session, as their access token is no longer valid within the backend. 

Writing this becomes a bit tricky as it relates to React:

- The Axios HTTP Interceptor resides in `backend-api.ts` -- a "POJO" -- and does not run within React's render lifecycle.
- The `PrivateRoute` route guard relies on React state to determine whether a user is authenticated (has a session stored).
- The HTTP Interceptor cannot reach out and update the React state `PrivateRoute` relies on for routing.
- Conversely, React cannot listen for changes to the session object as it does not reside within React. 

To summarize, it is trivial to clear the session through the HTTP Interceptor on HTTP 401 Responses. However, it is difficult to relay this information to React and the router.

**Comment by eric-horodyski** [comment]
Fixed with the updated `main` branch. The approach to sharing state inside and outside of React has been changed.
